src/main/java/com/ericsson/ei/erqueryservice/ERQueryService.java

```diff
@@ -70,16 +73,20 @@ public void setRest(RestOperations rest) {
      * @return ResponseEntity
      */
     public ResponseEntity<String> getEventDataById(String eventId) {
-        final String erUrl = URI.create(url.trim() + "/" + "{id}").normalize().toString();
-        log.debug("The URL to ER is: " + erUrl);
-
-        final Map<String, String> params = Collections.singletonMap("id", eventId);
-        log.trace("The ID parameter is set");
-        try {
-            final ResponseEntity<String> response = rest.getForEntity(erUrl, String.class, params);
-            log.trace("The response is : " + response.toString());
-        } catch (RestClientException e) {
-            log.error("Error occurred while executing REST GET to: " + erUrl + " for " + eventId, e);
+        if(StringUtils.isNotBlank(url)) {
+            final String erUrl = URI.create(url.trim() + "/" + "{id}").normalize().toString();
+            LOGGER.debug("The URL to ER is: " + erUrl);
+
+            final Map<String, String> params = Collections.singletonMap("id", eventId);
+            LOGGER.trace("The ID parameter is set");
+            try {
+                final ResponseEntity<String> response = rest.getForEntity(erUrl, String.class, params);
+                LOGGER.trace("The response is : " + response.toString());
+            } catch (RestClientException e) {
+                LOGGER.error("Error occurred while executing REST GET to: " + erUrl + " for " + eventId, e);
```

Comment:
These Errors (or Error informations) should be included in the ResponseEntity, if its returned in EI RestAPI responses.

Reply:
This comment applies to the other errors in this file which returns Responses to RestAPI.

Reply:
I have only added LOGGER.info("The URL to ER is not provided"); here.

Reply:
The Comment was about including the Error in the ResponseEntiy,, not Logger,,, Logger should be kept,, but include error into the ResponseEntity if its good information to user for clearing out what failed in the RestAPI call, if this was executed by a RestAPI request.

Reply:
Let it be and it will be handled by coming new story.